src/app/header/header.component.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  ViewChild,
+  AfterViewInit
+} from "@angular/core";
+import { AuthService } from "../auth.service";
+import { Router } from "@angular/router";
+
+@Component({
+  selector: "app-header",
+  templateUrl: "./header.component.html",
+  styleUrls: ["./header.component.css"]
+})
+export class HeaderComponent implements OnInit {
+  @Input() name: string;
+  @ViewChild("logIn", { static: false }) logIn;
```

Comment:
why are you using viewchild for showing login and logout button. You can make use of structural directives like *ngIf